#!/usr/bin/env python

PACKAGE = "people_follower"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('x_vel_gain', double_t, 0, 'The gain applied to distance to calculate x velocity', 0.1, 0.0)
gen.add('x_vel_min', double_t, 0, 'The min x velocity in [m/s]', 0.05, 0.0)
gen.add('x_vel_max', double_t, 0, 'The max x velocity in [m/s]', 0.25, 0.0)
gen.add('th_vel_gain', double_t, 0, 'The gain applied to angle difference to calculate angular velocity', 5, 0.0)
gen.add('th_vel_min', double_t, 0, 'The min angular velocity in [rad/s]', 0.26, 0.0)
gen.add('th_vel_max', double_t, 0, 'The max angular velocity in [rad/s]', 0.52, 0.0)
gen.add('d_thres', double_t, 0, 'The distance threshold', 0.5, 0)
gen.add('th_thres', double_t, 0, 'The angular differrence threshold', 0.05, 0)
gen.add('transform_timeout', double_t, 0, 'The timeout used asking for transformation from people frame to base frame', 0.1, 0)

exit(gen.generate(PACKAGE, PACKAGE, "PeopleFollower"))
